compiler/rustc_driver_impl/src/lib.rs

```diff
@@ -372,27 +372,23 @@ fn run_compiler(
 
         let handler = EarlyErrorHandler::new(sess.opts.error_format);
 
-        let should_stop = print_crate_info(
-            &handler,
-            &**compiler.codegen_backend(),
-            compiler.session(),
-            has_input,
-        );
+        if print_crate_info(&handler, &**compiler.codegen_backend(), compiler.session(), has_input)
+            == Compilation::Stop
+        {
+            return sess.compile_status();
+        }
 
         if !has_input {
-            if should_stop == Compilation::Continue {
-                handler.early_error("no input filename given")
-            }
-            return sess.compile_status();
+            handler.early_error("no input filename given");
```

Comment:
Doesn't this need `return sess.compile_status();`?

Reply:
No, because `early_error` is fatal. I could add a comment about that.